### ISSUE TITLE
Music takes very long to load

### ISSUE THREAD

**Issue by Ninjahague** [timusus/Shuttle#276]
**Shuttle version:**
 
2.02 Upgraded
 
**Device, OS:**
Galaxy S7, Android 7.0

**Description of bug:**

It takes like 30-60 seconds to fully open the music library

Music is on SD card
 
**Steps to reproduce:**
 
 1. Fully close shuttle with no music playing
 2. Open up shuttle through the widget or app icon
 3. Wait in sadness
 
**Expected outcome:**
 
Music should load instantaneously

**Comment by timusus** [comment]
@Ninjahague do you know what the make/model of your SD Card is? 

Related to / possible duplicate of #79

**Comment by Ninjahague** [comment]
This Model. I threw the amazon link down there

(P-SDU128U185EL-GE)

https://www.amazon.com/gp/aw/d/B01G26R7LA/ref=mp_s_a_1_2/136-1799559-3751901?ie=UTF8&qid=1516725383&sr=8-2&pi=AC_SX236_SY340_FMwebp_QL65&keywords=pny+128gb&dpPl=1&dpID=41TQx4c3LUL&ref=plSrch

**Comment by Dan33185** [comment]
Wondering if my issue might be related. I close out all apps, load Shuttle and it takes 15 or so seconds for my startup tab (artists) to load. Pulsar is almost instantaneous in the same environment. I have attached a video below as an example. Phone specs:

Galaxy J7 Prime
Unrooted
Android 7.0

https://youtu.be/1XXpQIHgft8

**Comment by Ninjahague** [comment]
I'm slightly curious if this could be due to the amount of folders/storage in the SD card itself and shuttle doesn't like it?

I currently have 76.06 gb on my SD card. 393 folders are dedicated to music based in one folder that is a total of 57.33 gb. the rest of the storage on my card being anime and such.

But im no means a software guy

**Comment by timusus** [comment]
My gut feeling it this is a combination of a Samsung bug and having lots of music on your device. When the app launches, it makes a bunch of SQL queries to gather information about your music. On some Samsung devices, if too many queries happen at the same time, there's a 'cursor window' issue, where it seems the device is complaining about too many queries happening, and waits until previous queries finish, before starting the next. This seems to have a sort of knock-on effect, which slows down all subsequent queries, so it takes a long time for Shuttle to finish those queries.

I don't really have a solution at this stage, and it's a very difficult one to investigate without being able to reproduce the issue.

**Comment by timusus** [comment]
Related to / possible duplicate of #79

**Comment by Ninjahague** [comment]
https://gfycat.com/BountifulBarrenJerboa

Just made a fancy video for you.

**Comment by Ninjahague** [comment]
📣 🤔 

Google Music opens instantly though, no problem though.

**Comment by timusus** [comment]
That doesn't really bear any relevance, unfortunately.

**Comment by Ninjahague** [comment]
darn lol. Anything else you would need then? I don't know what i'd do without shuttle. Google Play is cancer upon this earth and i need my album art

**Comment by timusus** [comment]
I don't know at this stage. You should come and join the Slack conversation, and perhaps I'll get you to provide some device logs at some point.

Can you post your device to Australia?

**Comment by Ninjahague** [comment]
How do i join this slack convo?. Also i dont know what you mean by posting my device to Australia lol sorry

**Comment by timusus** [comment]
Follow this link:
http://shuttle-slack-inviter.herokuapp.com/

I mean, if you can send your device to me, I'll have a better chance of fixing the bug! ;)

**Comment by Dan33185** [comment]
Is there a reason why an app like Pulsar would work as quick as it does, compared to what Shuttle is doing?

**Comment by timusus** [comment]
The reason is as simple as Pulsar is a different app, with a different codebase, that isn't encountering the same bug.

**Comment by timusus** [comment]
Closing as duplicate of #79. We can continue the discussion there.
